godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/String.kt

```diff
@@ -0,0 +1,22 @@
+@file:Suppress("unused", "MemberVisibilityCanBePrivate")
+
+package godot.core
+
+import godot.gdnative.*
+import kotlinx.cinterop.*
+
+
+fun String(mem: COpaquePointer) = mem.reinterpret<godot_string>().pointed.readValue().toKString()
+
+fun String.getRawMemory(memScope: MemScope): COpaquePointer =
+    cValue<godot_string> { godot_string_parse_utf8(this.ptr, this@getRawMemory) }.getPointer(memScope)
+
+internal fun String.toGDString(): CValue<godot_string> = cValue { godot_string_parse_utf8(this.ptr, this@toGDString) }
```

Comment:
Need to call `godot_string_destroy` when `godot_string` is not used anymore otherwise you will be leaking memory IIRC.

Reply:
Yeah sorry, I just reused the previous one without changing anything. I needed it for some functions in the coretypes